hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/impl/StatesImpl.java

```diff
@@ -56,6 +56,23 @@ public void updateChildren(final ServicesState sourceState) {
             final var state =
                     new RebuiltStateImpl<>(stateKey, children.aliases(), children.signedAt());
             return (StateBase) state;
+        } else if (stateKey.equals("SCHEDULES-BY-EQUALITY")) {
+            final var state =
+                    new InMemoryStateImpl<>(
+                            stateKey, children.schedules().byEquality(), children.signedAt());
+            return (StateBase) state;
+        } else if (stateKey.equals("SCHEDULES-BY-ID")) {
```

Comment:
Is `StatesImpl` an intermediate dummy-implementation? Maybe you can add a comment making this clear.

The if-else-construct should be replaced with a switch-expression IMO.

Reply:
yeah it was added as temporary solution. Added comment and changed to use switch . Thanks !